packages/amplify_authenticator/lib/amplify_authenticator.dart

```diff
@@ -204,26 +205,42 @@ class _AuthenticatorState extends State<Authenticator> {
   final AuthService _authService = AmplifyAuthService();
 
   late final StateMachineBloc _stateMachineBloc;
+  AmplifyConfig? _config;
 
   @override
   void initState() {
     super.initState();
     _stateMachineBloc = StateMachineBloc(_authService)
       ..authEvent.add(GetCurrentUser());
+    waitForConfiguration().then((value) {
+      _stateMachineBloc.configSink.add(value);
+    });
+  }
+
+  Future<AmplifyConfig> waitForConfiguration() async {
+    return await _authService.waitForConfiguration();
   }
 
   @override
   Widget build(BuildContext context) {
+    _stateMachineBloc.configStream.listen((event) {
```

Comment:
Should move to `initState` so it's not called on every build (will add multiple listeners).

Reply:
Done!